stdlib/public/core/KeyPath.swift

```diff
@@ -1951,7 +1951,8 @@ func _modifyAtWritableKeyPath_impl<Root, Value>(
       keyPath: _unsafeUncheckedDowncast(keyPath,
         to: ReferenceWritableKeyPath<Root, Value>.self))
   }
-  return keyPath._projectMutableAddress(from: &root)
+  let ptr = UnsafePointer<Root>(Builtin.unprotectedAddressOf(&root))
```

Comment:
Can this use the scoped `_withUnprotectedUnsafePointer` function rather than using the Builtin to get an unscoped pointer?

Reply:
yeah, that's a good idea